conversation_reconstruction/dataflow/write_page_states_to_spanner/dataflow_main.py

```diff
@@ -56,31 +62,43 @@ def run(known_args, pipeline_args):
 
   with beam.Pipeline(options=pipeline_options) as p:
     # Main Pipeline
-    p = (p | beam.io.ReadFromText(known_args.input_storage)
-         | beam.ParDo(WriteToSpanner(), known_args.spanner_instance, known_args.spanner_database,
-                      known_args.spanner_table, known_args.spanner_table_columns))
+    if known_args.input_storage is not None:
```

Comment:
You should fail with a readable error message if they are both specified, or if neither is specified. You should probably do this before the `with beam.Pipeline`